src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -369,6 +440,11 @@ public override async Task RunAsync()
 
             Utilities.PrintVersion();
 
+            if (ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
+            {
+                await StartInProc8AsChildProcessAsync();
+            }
```

Comment:
Yes. Good catch. This was one of the reason for the error I was getting because the net8 process failed and this code continued and tried to execute from the net6 exe (parent exe) context.